vb6/VisualBasic6Parser.g4

```diff
@@ -0,0 +1,845 @@
+		;
+
+seekStmt: SEEK WS valueStmt WS? COMMA WS? valueStmt;
+
+selectCaseStmt:
+	SELECT WS CASE WS valueStmt NEWLINE+ sC_Case* WS? END_SELECT;
+
+sC_Case:
+	CASE WS sC_Cond WS? (COLON? NEWLINE* | NEWLINE+) (
+		block NEWLINE+
+	)?;
+
+// ELSE first, so that it is not interpreted as a variable call
+sC_Cond:
+	ELSE										# caseCondElse
+	| sC_CondExpr (WS? COMMA WS? sC_CondExpr)*	# caseCondExpr;
+
+sC_CondExpr:
+	IS WS? comparisonOperator WS? valueStmt	# caseCondExprIs
+	| valueStmt								# caseCondExprValue
+	| valueStmt WS TO WS valueStmt			# caseCondExprTo;
+
+sendkeysStmt: SENDKEYS WS valueStmt (WS? COMMA WS? valueStmt)?;
+
+setattrStmt: SETATTR WS valueStmt WS? COMMA WS? valueStmt;
+
+setStmt: SET WS implicitCallStmt_InStmt WS? EQ WS? valueStmt;
+
+stopStmt: STOP;
+
+subStmt: (visibility WS)? (STATIC WS)? SUB WS ambiguousIdentifier (
+		WS? argList
+	)? NEWLINE+ (block NEWLINE+)? END_SUB;
+
+timeStmt: TIME WS? EQ WS? valueStmt;
+
+typeStmt: (visibility WS)? TYPE WS ambiguousIdentifier NEWLINE+ (
+		typeStmt_Element
+	)* END_TYPE;
+
+typeStmt_Element:
+	ambiguousIdentifier (WS? LPAREN (WS? subscripts)? WS? RPAREN)? (
+		WS asTypeClause
+	)? NEWLINE+;
+
+typeOfStmt: TYPEOF WS valueStmt (WS IS WS type)?;
+
+unloadStmt: UNLOAD WS valueStmt;
+
+unlockStmt:
+	UNLOCK WS valueStmt (
+		WS? COMMA WS? valueStmt (WS TO WS valueStmt)?
+	)?;
+
+// operator precedence is represented by rule order
+valueStmt:
+	literal															# vsLiteral
+	| LPAREN WS? valueStmt (WS? COMMA WS? valueStmt)* WS? RPAREN	# vsStruct
+	| NEW WS valueStmt												# vsNew
+	| typeOfStmt													# vsTypeOf
+	| ADDRESSOF WS valueStmt										# vsAddressOf
+	| implicitCallStmt_InStmt WS? ASSIGN WS? valueStmt				# vsAssign
+	| valueStmt WS? POW WS? valueStmt								# vsPow
+	| MINUS WS? valueStmt											# vsNegation
+	| PLUS WS? valueStmt											# vsPlus
+	| valueStmt WS? DIV WS? valueStmt								# vsDiv
+	| valueStmt WS? MULT WS? valueStmt								# vsMult
+	| valueStmt WS? MOD WS? valueStmt								# vsMod
+	| valueStmt WS? PLUS WS? valueStmt								# vsAdd
+	| valueStmt WS? MINUS WS? valueStmt								# vsMinus
+	| valueStmt WS? AMPERSAND WS? valueStmt							# vsAmp
+	| valueStmt WS? EQ WS? valueStmt								# vsEq
+	| valueStmt WS? NEQ WS? valueStmt								# vsNeq
+	| valueStmt WS? LT WS? valueStmt								# vsLt
+	| valueStmt WS? GT WS? valueStmt								# vsGt
+	| valueStmt WS? LEQ WS? valueStmt								# vsLeq
+	| valueStmt WS? GEQ WS? valueStmt								# vsGeq
+	| valueStmt WS LIKE WS valueStmt								# vsLike
+	| valueStmt WS IS WS valueStmt									# vsIs
+	| NOT (WS valueStmt | LPAREN WS? valueStmt WS? RPAREN)			# vsNot
+	| valueStmt WS? AND WS? valueStmt								# vsAnd
+	| valueStmt WS? OR WS? valueStmt								# vsOr
+	| valueStmt WS? XOR WS? valueStmt								# vsXor
+	| valueStmt WS? EQV WS? valueStmt								# vsEqv
+	| valueStmt WS? IMP WS? valueStmt								# vsImp
+	| implicitCallStmt_InStmt										# vsICS
+	| midStmt														# vsMid;
```

Comment:
Operators precedence is incorrect. For instance, in the following code division is parsed before multiplication, but should be otherwise.

```
a * b / c
```

See https://docs.microsoft.com/en-us/dotnet/visual-basic/language-reference/operators/operator-precedence#precedence-rules

Correct grammar rule looks like this:

```antlr
    : literal                                                       # vsLiteral
    | LPAREN WS? valueStmt (WS? COMMA WS? valueStmt)* WS? RPAREN    # vsStruct
    | NEW WS valueStmt                                              # vsNew
    | typeOfStmt                                                    # vsTypeOf
    | ADDRESSOF WS valueStmt                                        # vsAddressOf
    | implicitCallStmt_InStmt WS? ASSIGN WS? valueStmt              # vsAssign
    | valueStmt WS? POW WS? valueStmt                               # vsPow
    | (PLUS | MINUS) WS? valueStmt                                  # vsPlusMinus
    | valueStmt WS? (MULT | DIV) WS? valueStmt                      # vsMultDiv
    | valueStmt WS? (IDIV) WS? valueStmt                            # vsIDiv
    | valueStmt WS? MOD WS? valueStmt                               # vsMod
    | valueStmt WS? (PLUS | MINUS) WS? valueStmt                    # vsAddSub
    | valueStmt WS? AMPERSAND WS? valueStmt                         # vsAmp
    | valueStmt WS? (EQ | NEQ | LT | GT | LEQ | GEQ | LIKE | IS) WS? valueStmt # vsComp
    | NOT (WS valueStmt | LPAREN WS? valueStmt WS? RPAREN)          # vsNot
    | valueStmt WS? AND WS? valueStmt                               # vsAnd
    | valueStmt WS? OR WS? valueStmt                                # vsOr
    | valueStmt WS? XOR WS? valueStmt                               # vsXor
    | valueStmt WS? EQV WS? valueStmt                               # vsEqv
    | valueStmt WS? IMP WS? valueStmt                               # vsImp
    | implicitCallStmt_InStmt                                       # vsICS
    | midStmt                                                       # vsMid
    ;
```